### DISCUSSION TITLE
How to get subtitles?

### DISCUSSION THREAD

**Discussion by pvonmoradi** [iptv-org/community#254] [Q&A] [answered]
The project provides `.m3u8` streams which are playable in apps like VLC.  
Now, in TV mode, I know some channels provide subtitles alongside their stream (like ZDF, RTL). 
I don't know much about IPTV protocol. Is that subtitle file also incorporated into the stream?   

Having a synced subtitle would be great for purposes like language learning.  
Thanks for this awesome project.

**Comment by BellezaEmporium** [comment, marked as answer]
For players that support it, you just need to check the "subs" option when you watch the program.

**Reply by pvonmoradi** [reply]
@LaneSh4d0w I use VLC but there were no subs for channels I tested. Do you know a channel which has subs integrated so I can test it?

**Reply by BellezaEmporium** [reply]
Try Canal RCN : http://186.101.216.211:8000/play/a09k

**Reply by pvonmoradi** [reply]
@LaneSh4d0w Thanks! Yeah it works. One more thing: for checking whether a stream has subs, I can simply ffprobe | grep for "Closed Captions", right? https://dpaste.com/G65JRSH3B

**Reply by BellezaEmporium** [reply]
On this one, there's 4 different captions, so yes, you can check out for something called Closed Captions or Hearing Impaired Captions

**Reply by pvonmoradi** [reply]
For the record, I just checked all "United Kingdom", "Switzerland" and "Germany" links. None of them have any subtitles.  
(except https://liveprodeuwest.global.ssl.fastly.net/btv/desktop/eu_event.m3u8 which is not helpful actually)

**Reply by BellezaEmporium** [reply]
Yes, we do not have a lot of subtitled streams. Those aren't on our resort, there's not a lot of subtitled streams available on the internet.

**Reply by pvonmoradi** [reply]
For anyone wondering, here are the US links (count=136) that have subtitles:  
 https://paste.ubuntu.com/p/cXFytCSTkj/
